nova/scheduler/filters/shard_filter.py

```diff
@@ -28,6 +32,9 @@
 CONF = nova.conf.CONF
 
 _SERVICE_AUTH = None
+GARDENER_PREFIX = "kubernetes.io-cluster-shoot--garden--"
```

Comment:
I checked in eu-de-2, and I see there some entries starting just with `kubernetes.io-cluster-shoot--hcm-eng` .
I would shorten the prefix check to `kubernetes.io-cluster-`, as I don't think we gain much by getting more specific.